### DISCUSSION TITLE
Enhancement: reload assets + "images" frame viewer"

### DISCUSSION THREAD

**Discussion by Tayo-Fox** [FunkinCrew/Funkin#1598] [Ideas]
2 additions to the "7" debug menu

1. a button to reload all currently loaded assets, to help people who mod the game.
currently some assets are reloaded per level load like boyfriend, daddy dearest, pico, etc.
a set of assets require the whole game to be reloaded, like note assets, audio, etc
(the reload audio button doesnt seem to actually reload the audio)
having a button to reload these on the fly would be a big help to anyone who is updating and looking at changes in assets on the fly.

2. animation viewer.
this would also be a quality of life thing for modders, a simple display of what model is currently loaded, and what frame of animation it is displaying (or to play the whole animation)
ideas for keys would be like
W/S to change model
A/D to change animation/frame
Space/Q to toggle animation or frame by frame
E/R to reload XML/PNG

just tossing those ideas out there.
i imagine these would help people with creating assets for mods
(or at least help people like me who don't have adobe animate and modify the xml file with a text editor lol)

**Comment by AbnormalPoof** [comment]
the animation viewer kinda exists

build the game on debug and press 8 during a song

**Reply by ghost** [reply]
wait that actually exists??

**Comment by ninjamuffin99** [comment]
animation viewer will be in week 8 update, I already fully made it and use it

**Reply by ACrazyTown** [reply]
will it have the option to adjust offsets in-game? cuz currently i'm using a seperate tool for offsets and it's kinda tedious

**Reply by AbnormalPoof** [reply]
ayo?????????/

**Reply by AbnormalPoof** [reply]
is it different from animation debug or the same

**Reply by ninjamuffin99** [reply]
@ACrazyTown animation offsets can be made ingame already, but might only be available in debug mode

@AbnormalPoof same, but muuuch better

**Reply by AbnormalPoof** [reply]
> @AbnormalPoof same, but muuuch better

sounds promising
